sdk/metric/metricdata/data.go

```diff
@@ -122,9 +122,24 @@ type HistogramDataPoint struct {
 	BucketCounts []uint64
 
 	// Min is the minimum value recorded. (optional)
-	Min *float64 `json:",omitempty"`
+	Min Extrema
 	// Max is the maximum value recorded. (optional)
-	Max *float64 `json:",omitempty"`
+	Max Extrema
 	// Sum is the sum of the values recorded.
 	Sum float64
 }
+
+// Extrema are the minimum or maximum values of a data set.
+type Extrema struct {
+	// Value is the extrema value.
+	Value float64
+
+	// Valid is true if Value has been assigned a value. It is false if Value
+	// is the zero-value.
+	Valid bool
```

Comment:
Should this be non-exported with an accessor like `IsValid()`?  I can't imagine why anyone would change this value but by exporting the field we leave that option open.

Reply:
I think by including the `NewExtrema` function having accessors seems to make sense. I think we would want to unify on the accessor methods instead of using fields at all though. Otherwise, part of the type's functionality will be accessed one way and the other another.

What do you think about:

```go
func (Extrema) Value() (float64, bool)
```

With the second returned value being `true` if the value is defined and `false` if not?

Reply:
I don't think there would be a need for a setter. A user would just create a new `Extrema`.

Reply:
I agree that the value can be read-only and having a single `Value() (float64, bool)` accessor sounds good to me.